src/toil_lib/tools/variant_manipulation.py

```diff
@@ -0,0 +1,351 @@
+                parameters=command,
+                tool='quay.io/ucsc_cgl/gatk:3.5--dba6dae49156168a909c43330350c6161dc7ecc2',
+                inputs=inputs.keys(),
+                outputs={'output.vcf': None})
+
+    return job.fileStore.writeGlobalFile(os.path.join(work_dir, 'output.vcf'))
+
+
+def gatk_variant_filtration(job, mode, vcf_id, ref_fasta, ref_fai, ref_dict):
+    """
+    Filters VCF file using GATK VariantFiltration. The filter expressions are based on GATK variant annotation features.
+
+    SNP Filter Expression
+    QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0
+
+    INDEL Filter Expression
+    QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0
+
+    :param JobFunctionWrappingJob job: passed automatically by Toil
+    :param str mode: variant type (SNP or INDEL)
+    :param str vcf_id: FileStoreID for input VCF file
+    :param str ref_fasta: FileStoreID for reference genome fasta
+    :param str ref_fai: FileStoreID for reference genome index file
+    :param str ref_dict: FileStoreID for reference genome sequence dictionary file
+    :return: FileStoreID for filtered VCF file
+    :rtype: str
+    """
+    mode = mode.upper()
+    job.fileStore.logToMaster('GATK VariantFiltration: %s' % mode)
+
+    inputs = {'genome.fa': ref_fasta,
+              'genome.fa.fai': ref_fai,
+              'genome.dict': ref_dict,
+              'input.vcf': vcf_id}
+
+    work_dir = job.fileStore.getLocalTempDir()
+    for name, file_store_id in inputs.iteritems():
+        job.fileStore.readGlobalFile(file_store_id, os.path.join(work_dir, name))
+
+    # GATK hard filters:
+    # https://software.broadinstitute.org/gatk/documentation/article?id=2806
+    if mode == 'SNP':
+        expression = '"QD < 2.0 || FS > 60.0 || MQ < 40.0 || ' \
+                     'MQRankSum < -12.5 || ReadPosRankSum < -8.0"'
+
+    elif mode == 'INDEL':
+        expression = '"QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0"'
+
+    else:
+        raise ValueError('Variant filter modes can be SNP or INDEL, got %s' % mode)
+
+    command = ['-T', 'VariantFiltration',
+               '-R', 'genome.fa',
+               '-V', 'input.vcf',
+               '--filterExpression', expression,
+               '--filterName', 'GATK_Germline_Hard_Filter_%s' % mode,   # Documents filter in header
+               '-o', 'filtered_variants.vcf']
+
+    docker_call(work_dir=work_dir,
+                env={'JAVA_OPTS': '-Djava.io.tmpdir=/data/ -Xmx{}'.format(job.memory)},
+                parameters=command,
+                tool='quay.io/ucsc_cgl/gatk:3.5--dba6dae49156168a909c43330350c6161dc7ecc2',
+                inputs=inputs.keys(),
+                outputs={'filtered_variants.vcf': None})
+
+    # Fix extra quotation marks in FILTER line
+    output = os.path.join(work_dir, 'filtered_variants.vcf')
+    sed_cmd = 's/"{}"/{}/'.format(expression, expression)
+    subprocess.call(['sed', '-i', sed_cmd, output])
```

Comment:
Not sure if we care, but this usage of sed is not cross platform compatible. BSD sed requires an extension to be passed to `-i`. Long term, it might be preferable to move this into Python.